src/main/java/mate/academy/rickandmorty/service/impl/CharacterServiceImpl.java

```diff
@@ -0,0 +1,35 @@
+package mate.academy.rickandmorty.service.impl;
+
+import java.util.List;
+import java.util.Random;
+import lombok.RequiredArgsConstructor;
+import mate.academy.rickandmorty.dto.internal.InternalCharacterDto;
+import mate.academy.rickandmorty.exception.CharacterNotFoundException;
+import mate.academy.rickandmorty.mapper.CharacterMapper;
+import mate.academy.rickandmorty.repository.CharacterRepository;
+import mate.academy.rickandmorty.service.CharacterService;
+import org.springframework.stereotype.Service;
+
+@RequiredArgsConstructor
+@Service
+public class CharacterServiceImpl implements CharacterService {
+    private final CharacterRepository characterRepository;
+    private final CharacterMapper characterMapper;
+    private final Random random;
+
+    @Override
+    public InternalCharacterDto getRandomlyCharacter() {
+        Long randomId = random.nextLong(characterRepository.count()) + 1;
+        return characterMapper.toDto(characterRepository.findById(randomId)
+                .orElseThrow(()
+                        -> new CharacterNotFoundException("Can`t find a character by id: "
+                        + randomId)));
+    }
+
+    @Override
+    public List<InternalCharacterDto> getCharacterByName(String name) {
+        return characterRepository.findAllByName(name)
```

Comment:
```suggestion
        return characterRepository.findAllByNameIn(name)
```
![image](https://github.com/mate-academy/jv-rick-and-morty/assets/164332253/64482660-a3b0-486f-a6c3-04f4aff5d53a)
Let's say you have Rick in DB. If I call getCharacterByName("ck"), I want it to return Rick, Now your method will return Rick only if I call getCharacterByName("Rick")